samcli/cli/types.py

```diff
@@ -388,5 +388,26 @@ def convert(self, value, param, ctx):
         """
         result = self.transformer.transform(value, param, ctx)
         if not result:
-            raise click.BadParameter(f"{param.opts[0]} needs to be a valid ECR URI")
+            raise click.BadParameter(f"Invalid Image Repository ECR URI: {value}")
         return value
+
+
+class ImageRepositoriesType(click.ParamType):
+    """
+    Custom Parameter Type for Multi valued Image Repositories option.
+    """
+
+    name = ""
+
+    def convert(self, value, param, ctx):
+        key_value_pair = value.split("=")
+        if len(key_value_pair) != 2:
+            raise click.BadParameter(
+                f"{param.opts[0]} is not a valid format, it needs to be of the form function_logical_id=ECR_URI"
+            )
+        else:
+            key = key_value_pair[0]
+            _value = key_value_pair[1]
+            if not is_ecr_url(_value):
+                raise click.BadParameter(f"{param.opts[0]} needs to have valid ECR URI as value")
+            return {key: _value}
```

Comment:
this is a specific way of click operates with multi value options, we can do the other way but its not native click. ( this is how --parameter-overrides work).

I still prefer one, because its native impl and better representation in configuration file.